File: data_structures_by_python/linked_list/singly_linked_list/delete_at_any_position.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None


def delete_at_any_pos(head, idx):
    tmp = head
    for _ in range(1, idx):
        tmp = tmp.next
    delete_node = tmp.next
    tmp.next = delete_node.next
    del delete_node


def print_linked_list(head):
    tmp = head
    while tmp is not None:
        print(tmp.val, end=" ")
        tmp = tmp.next
    print()


head = Node(10)
a = Node(20)
tail = Node(30)
head.next = a
a.next = tail
delete_at_any_pos(head, 2)
print_linked_list(head)

File: data_structures_by_c++/linked_list/singly_linked_list/delete_at_any_position.cpp
#include <bits/stdc++.h>
using namespace std;

class Node
{
    public:
        int val;
        Node* next;
        Node(int val)
        {
            this->val = val;
            this->next = NULL;
        }
};

void delete_at_any_pos(Node* head, int idx)
{
    Node* tmp = head;
    for (int i = 1; i < idx; i++)
    {
        tmp = tmp->next;
    }
    Node* deleteNode = tmp->next;
    tmp->next = deleteNode->next;
    delete deleteNode;
}

void print_linked_list(Node* head)
{
    Node* tmp = head;
    while(tmp != NULL)
    {
        cout << tmp->val << " ";
        tmp = tmp->next;
    }
    cout << endl;
}

int main()
{
    Node* head = new Node(10);
    Node* a = new Node(20);
    Node* tail = new Node(30);

    head->next = a;
    a->next = tail;

    delete_at_any_pos(head, 2);
    print_linked_list(head);
}
